adam/curriculum/pursuit_curriculum.py

```diff
@@ -0,0 +1,118 @@
+        a scene where the utterance doesn't match the situation and perception (e.g. hearing "ball" while seeing a cup).
+        num_objects_in_instance: number of objects in each instance
+    """
+    if num_noise_instances > num_instances:
+        raise RuntimeError("Cannot have more noise than regular exemplars")
+
+    target_objects = [BALL, CHAIR, PERSON, TABLE, DOG, BIRD, BOX]
+    noise_object_variables = [
+        standard_object("obj-" + str(idx)) for idx in range(num_objects_in_instance)
+    ]
+
+    # A template that is used to replace situations and perceptions (not linguistic description) in noise instances
+    noise_template = Phase1SituationTemplate(
+        "simple_pursuit-noise",
+        salient_object_variables=[noise_object_variables[0]],
+        background_object_variables=noise_object_variables[1:],
+    )
+
+    all_instances = []
+    # Generate phase_1 instance groups for each template (i.e each target word)
+    for target_object in target_objects:
+        target_object_variable = object_variable(
+            target_object.handle + "-target", target_object
+        )
+        # For each target object, create a template with specific a target object in each to create learning instances.
+        # There is one object (e.g. Ball) across all instances while the other objects vary. Hence, the target object is
+        # a salient object (used for the linguistic description) while the remaining objects are background objects.
+        object_is_present_template = Phase1SituationTemplate(
+            "simple_pursuit",
+            salient_object_variables=[target_object_variable],
+            background_object_variables=noise_object_variables[:-1],
+        )
+        non_noise_instances = list(
+            phase1_instances(
+                "simple_pursuit_curriculum",
+                sampled(
+                    object_is_present_template,
+                    max_to_sample=num_instances - num_noise_instances,
+                    chooser=PHASE1_CHOOSER,
+                    ontology=GAILA_PHASE_1_ONTOLOGY,
+                ),
+            ).instances()
+        )
+        all_instances.extend(non_noise_instances)
+
+        # Create instances for noise
+        noise_instances = phase1_instances(
+            "simple_pursuit_curriculum",
+            sampled(
+                noise_template,
+                max_to_sample=num_noise_instances,
+                chooser=PHASE1_CHOOSER,
+                ontology=GAILA_PHASE_1_ONTOLOGY,
+            ),
+        ).instances()
+        for (situation, _, perception) in noise_instances:
+            # For each instance to be replaced by a noisy instance, we keep the correct utterance, but
+            # replace the situation and perception. We do this to test the model's tolerance to varying degrees of
+            # noise that reflects noisy instances in the real-world.
+            linguistic_description = PHASE1_CHOOSER.choice(non_noise_instances)[1]
```

Comment:
If I am reading this correctly, the linguistic descriptions for the non-noise instances for a given object type are always the same (the name of the object), so choosing randomly is a bit of a red herring.  I think you could put this outside this `for` loop:

```
# [1] is the index of the linguistic description in an instance
# It doesn't matter which non-noise instance is chosen
# because they all have the object type name as their linguistic description.
target_object_name = first(non_noise_instances)[1]
```

and then here you can do

```
linguistic_description = target_object_name
```